systests/src/utils/interface_helper.rs

```diff
@@ -0,0 +1,70 @@
+use std::net::IpAddr;
+
+use super::process::run_command;
+
+pub struct InterfaceHelper {
+    next_tun_num: u32,
+}
+
+impl InterfaceHelper {
+    pub fn new() -> Self {
+        Self { next_tun_num: 10 }
+    }
+
+    pub fn new_ifc_name(&mut self) -> String {
+        let tun_name = format!("tun{}", self.next_tun_num);
+        self.next_tun_num += 1;
+        tun_name
+    }
+
+    pub fn configure_ifc(ifc_name: &str, ip: IpAddr) {
+        run_command(
```

Comment:
I guess this wouldn't work running the tests on macOS because it uses `ifconfig` for interface management. as far as I know there is no one cross-platform crate for doing so, only retrieving the configuration..